crates/nu-protocol/src/eval_const.rs

```diff
@@ -259,39 +263,51 @@ pub(crate) fn create_nu_constant(engine_state: &EngineState, span: Span) -> Valu
     Value::record(record, span)
 }
 
-pub fn get_vendor_autoload_dir(engine_state: &EngineState) -> Option<PathBuf> {
+pub fn get_vendor_autoload_dir(engine_state: &EngineState) -> Vec<PathBuf> {
+    // TODO: update docs
     // pseudo code
     // if env var NU_VENDOR_AUTOLOAD_DIR is set, in any platform, use it
     // if not, if windows, use ALLUSERPROFILE\nushell\vendor\autoload
     // if not, if non-windows, if env var PREFIX is set, use PREFIX/share/nushell/vendor/autoload
     // if not, use the default /usr/share/nushell/vendor/autoload
 
-    // check to see if NU_VENDOR_AUTOLOAD_DIR env var is set, if not, use the default
-    Some(
-        option_env!("NU_VENDOR_AUTOLOAD_DIR")
-            .map(String::from)
-            .unwrap_or_else(|| {
-                if cfg!(windows) {
-                    let all_user_profile = match engine_state.get_env_var("ALLUSERPROFILE") {
-                        Some(v) => format!(
-                            "{}\\nushell\\vendor\\autoload",
-                            v.coerce_string().unwrap_or("C:\\ProgramData".into())
-                        ),
-                        None => "C:\\ProgramData\\nushell\\vendor\\autoload".into(),
-                    };
-                    all_user_profile
-                } else {
-                    // In non-Windows environments, if NU_VENDOR_AUTOLOAD_DIR is not set
-                    // check to see if PREFIX env var is set, and use it as PREFIX/nushell/vendor/autoload
-                    // otherwise default to /usr/share/nushell/vendor/autoload
-                    option_env!("PREFIX").map(String::from).map_or_else(
-                        || "/usr/local/share/nushell/vendor/autoload".into(),
-                        |prefix| format!("{}/share/nushell/vendor/autoload", prefix),
-                    )
-                }
-            })
-            .into(),
-    )
+    let into_autoload_path_fn = |mut path: PathBuf| {
+        path.push("nushell");
+        path.push("vendor");
+        path.push("autoload");
+        path
+    };
+    let mut dirs = Vec::new();
+
+    // platform base directies
+    // TODO: We're handling windows and not windows at the moment. using XDG_DATA_DIRS and falling back to `/usr/(local/)share` on MacOS might not be correct from my reading of the Apple developer documentation.
```

Comment:
I'd like to receive some feedback on this. What do y'all think? https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1 is the relevant documenation ftr

Reply:
This is where it gets a bit messy. We decided to go with `~/Library/Application Support` for our config, so it would kind of make sense to go with `/Library/Application Support` for the system level config too.

However, most users on macOS of TUI apps expect them to basically use XDG dirs like `~/.config` and I've seen a lot of people set `XDG_CONFIG_HOME` after we started supporting that to avoid `~/Library`. I think the perception is that `~/Library` is the right place for macOS graphical applications to put things, but that the Terminal should behave like any other Unix-like environment

I think just for consistency for now we might as well go with `/Library/Application Support` and *also* support `XDG_DATA_DIRS`. I've also heard from Windows folks that they would like to be able to configure stuff as well and they don't mind using the XDG spec environment variables to do that, but that kind of creates a problem for `XDG_DATA_DIRS` because it's colon separated, we'd have to be non-standard and allow semicolon to be used like Windows `Path`.

And I think compile time configuration is important to allow Homebrew or whatever else to do what they want. I'm not sure if they rely on setting `XDG_DATA_DIRS` but I do know they set `PREFIX` at compile time.

Reply:
At least for the autoload dirs, we don't have to decide on either but can instead just go for both without any trouble? The more, the merrier. We just have to decide on a precedence.

> And I think compile time configuration is important to allow Homebrew or whatever else to do what they want. I'm not sure if they rely on setting XDG_DATA_DIRS but I do know they set PREFIX at compile time.

So, what do you suggest here? https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables says 

> If $XDG_DATA_DIRS is either not set or empty, a value equal to /usr/local/share/:/usr/share/ should be used.

Should we just replace `/usr` with `PREFIX` for both of those? Add an additional one with `PREFIX`, and dedupe the list in the end? I think that would be my preferred option.

Reply:
I suppose we wouldn't want to do `/usr/local/local` if `PREFIX = /usr/local` though, so it might be reasonable to just special case that

But I think that does sound fine to me